clients/ctl/admin-ui/src/features/dataset/DatabaseConnectForm.tsx

```diff
@@ -162,7 +168,12 @@ const DatabaseConnectForm = () => {
       return;
     }
 
-    const createResult = await create(generateResult.dataset);
+    // Usually only one dataset needs to be created, but create them all just in case.
+    const createResults = await Promise.all(
+      generateResult.datasets.map((dataset) => create(dataset))
```

Comment:
I guess we could do an upsert to create everyone in one go instead. but I think it's probably okay to leave as-is since it's usually only one